base_local_planner/src/goal_functions.cpp

```diff
@@ -177,9 +177,7 @@ namespace base_local_planner {
 
     const geometry_msgs::PoseStamped& plan_goal_pose = global_plan.back();
     try{
-      geometry_msgs::TransformStamped transform = tf.lookupTransform(global_frame, ros::Time(),
-                         plan_goal_pose.header.frame_id, plan_goal_pose.header.stamp,
-                         plan_goal_pose.header.frame_id, ros::Duration(0.5));
+      geometry_msgs::TransformStamped transform = tf.lookupTransform(global_frame, plan_goal_pose.header.frame_id, time, ros::Duration(0.5));
```

Comment:
why is ros::Time() a problem here? that should simply give us the latest transform - we don't even appear to be ever using the timestamp of the goal_pose returned, so?

Reply:
We don't use the timestamp of returned goal_pose, but the timestamp is used to define transformation itself. And it's better to be consistent with the rest of computations. That is why I still propose to pass timestamp here and pass robot's pose timestamp (in LatchedStopRotateController).